mypy/checker.py

```diff
@@ -148,6 +152,12 @@ def visit_file(self, file_node: MypyFile, path: str) -> None:
         self.weak_opts = file_node.weak_opts
         self.enter_partial_types()
         self.is_typeshed_stub = self.errors.is_typeshed_file(path)
+        if self.options.strict_optional_whitelist is None:
+            self.suppress_none_errors = False
+        else:
+            self.suppress_none_errors = not any(fnmatch.fnmatch(path, pattern)
```

Comment:
Not in the stdlib -- normally one would use pathlib, which has a glob that supports **, but we don't want to generate the list of all matching files. So leave it be.